muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/MuzzleReferencesAccessor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.muzzle;
+
+import static java.util.Collections.emptyMap;
+
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.muzzle.ClassRef;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Map;
+
+/**
+ * This class is used to access {@code getMuzzleReferences} method from the {@link
+ * InstrumentationModule} class. That method is not visible in the source code and instead is
+ * generated automatically during compilation by the {@code
+ * io.opentelemetry.instrumentation.javaagent-codegen} Gradle plugin.
+ */
+class MuzzleReferencesAccessor {
+  private static final ClassValue<MethodHandle> getMuzzleReferences =
+      new ClassValue<MethodHandle>() {
```

Comment:
Now that I think of it we don't really need to use `ClassValue`s and `MethodHandle`s - the `getMuzzleReferences()` will be called exactly once per `InstrumentationModule` implementation, so there's no need to cache the result here (it's already cached in the `MuzzleMatcher`).

Reply:
Are you sure? Will be happy to remove this cache!

Reply:
Yep -- the only place (`MuzzleMatcher`) that constructs `ReferenceMatcher` in the agent uses an `AtomicBoolean initialized` field to make sure it only does it once (building a giant map of references can get expensive).

Reply:
What about `io.opentelemetry.javaagent.tooling.muzzle.ClassLoaderMatcher#matchesAll`?

Reply:
Also, I will try Lauri idea from https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4087#discussion_r707110186 anyway later on. That will probably remove this whole class.

Reply:
That one is only used in the gradle muzzle-check plugin, it doesn't affect the agent at all.